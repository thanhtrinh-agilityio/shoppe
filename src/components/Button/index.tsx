import { memo, ReactNode } from 'react'
import {
  Button as ButtonChakra,
  ButtonProps as ButtonPropsChakra,
} from '@chakra-ui/react'

interface ButtonProps extends ButtonPropsChakra {
  children: ReactNode
  variant: 'primary'
  size: 'default' | 'medium' | 'base' | 'small'
}

const Button = ({ children, variant, ...rest }: ButtonProps) => {
  return (
    <ButtonChakra variant={variant} {...rest}>
      {children}
    </ButtonChakra>
  )
}

export default memo(Button)
