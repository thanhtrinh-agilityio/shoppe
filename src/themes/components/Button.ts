import { ComponentStyleConfig } from '@chakra-ui/react'

export const Button: ComponentStyleConfig = {
  sizes: {
    small: {
      height: '25px',
      width: '25px',
    },
    base: {
      height: '53px',
      width: '53px',
    },
    default: {
      width: '353px',
      height: '53px',
    },
    medium: {
      height: '53px',
      width: '150px',
    },
  },
  variants: {
    primary: {
      textTransform: 'uppercase',
      bgColor: 'light',
      textColor: 'dark',
      borderRadius: '4px',
      border: '1px solid #000000',
      fontWeight: 'bold',
      fontSize: '16px',
    },
    default: {
      border: 'none',
    },
  },
  defaultProps: {
    variant: 'default',
    size: 'base',
  },
}
