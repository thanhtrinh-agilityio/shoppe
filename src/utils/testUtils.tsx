import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
