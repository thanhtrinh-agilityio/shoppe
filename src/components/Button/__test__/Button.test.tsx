import { fireEvent, render, screen } from '@utils/testUtils'
import Button from '@components/Button'

describe('Button Component', () => {
  const handleClick = jest.fn()

  const mockProps = {
    children: 'Add To Cart',
    onClick: handleClick,
  }
  it('Should show match Button DOM Snapshot', () => {
    const { container } = render(
      <Button variant="primary" size="default" {...mockProps} />,
    )

    expect(container).toMatchSnapshot()
  })

  it('Should call onClick on button ', () => {
    render(<Button variant="primary" size="default" {...mockProps} />)

    const button = screen.getByRole('button')

    fireEvent.click(button)

    expect(handleClick).toBeCalled()
    expect(handleClick).toBeCalledTimes(1)
  })
})
