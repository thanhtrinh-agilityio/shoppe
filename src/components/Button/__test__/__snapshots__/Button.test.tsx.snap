// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should show match Button DOM Snapshot 1`] = `
<div>
  <button
    class="chakra-button css-1a6r7m5"
    type="button"
  >
    Add To Cart
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
